lineapy/editors/ipython.py

```diff
@@ -99,16 +105,28 @@ def input_transformer_post(lines: List[str]) -> List[str]:
     # If we have just started, first start everything up
     if isinstance(STATE, StartedState):
         configure_logging()
-        db = RelationalLineaDB.from_environment(STATE.db_url)
+
+        db = RelationalLineaDB.from_config(options)
         # pass in globals from ipython so that `get_ipython()` works
         # and things like `!cat df.csv` work in the notebook
         ipython_globals = STATE.ipython.user_global_ns
         tracer = Tracer(
             db, SessionType.JUPYTER, STATE.session_name, ipython_globals
         )
 
-        STATE = CellsExecutedState(tracer, code=code)
+        STATE = CellsExecutedState(STATE.ipython, tracer, code=code)
     else:
+        if STATE.tracer.db.url != options.database_url:
+            STATE.ipython.reset()  # TODO: double check we are not over deleting variables
+            configure_logging()
+            db = RelationalLineaDB.from_config(options)
+            tracer = Tracer(
+                db,
+                SessionType.JUPYTER,
+                session_name,
+                STATE.ipython.user_global_ns,
+            )
+            STATE = CellsExecutedState(STATE.ipython, tracer, code=code)
```

Comment:
I need to look at this again in detail but I don't want to hold off your eng work.